File: Semester 5/Data Managment/Lab6/src/main/java/org/example/Klasa.java
package org.example;
import jakarta.persistence.*;
import java.util.List;
import java.io.Serializable;

@Entity
public class Klasa implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nazwa;
    private int poziom;
    @OneToOne(cascade = CascadeType.ALL)
    private Nauczyciel nauczyciel;
    @OneToMany(cascade = CascadeType.ALL)
    private List<Uczen> uczniowie;
    @OneToMany(cascade = CascadeType.ALL)
    private List<Przedmiot> przedmioty;

    public Nauczyciel getNauczyciel() {return nauczyciel;}
    public void setNauczyciel(Nauczyciel nauczyciel) {this.nauczyciel = nauczyciel;}

    public List<Uczen> getUczniowie() {return uczniowie;}
    public void setUczniowie(List<Uczen> uczniowie) {this.uczniowie = uczniowie;}

    public List<Przedmiot> getPrzedmioty() {return przedmioty;}
    public void setPrzedmioty(List<Przedmiot> przedmioty) {this.przedmioty = przedmioty;}

    public Long getId() {return id;}
    public void setId(Long id) {this.id = id;}

    public String getNazwa() {return nazwa;}
    public void setNazwa(String nazwa) {this.nazwa = nazwa;}

    public int getPoziom() {return poziom;}
    public void setPoziom(int poziom) {this.poziom = poziom;}
}

File: Semester 5/Data Managment/Lab7/src/Entity/Klasa.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

/**
 * @ORM\Entity
 */
class Klasa
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $nazwa;

    /**
     * @ORM\Column(type="integer")
     */
    private $poziom;

    /**
     * @ORM\OneToOne(targetEntity="Nauczyciel", cascade={"persist", "remove"})
     * @ORM\JoinColumn(name="nauczyciel_id", referencedColumnName="id")
     */
    private $nauczyciel;

    /**
     * @ORM\OneToMany(targetEntity="Uczen", mappedBy="klasa", cascade={"persist", "remove"})
     */
    private $uczniowie;

    /**
     * @ORM\OneToMany(targetEntity="Przedmiot", mappedBy="klasa", cascade={"persist", "remove"})
     */
    private $przedmioty;

    public function __construct(string $nazwa, int $poziom)
    {
        $this->nazwa = $nazwa;
        $this->poziom = $poziom;
        $this->uczniowie = new ArrayCollection();
        $this->przedmioty = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): void
    {
        $this->id = $id;
    }

    public function getNazwa(): ?string
    {
        return $this->nazwa;
    }

    public function setNazwa(string $nazwa): void
    {
        $this->nazwa = $nazwa;
    }

    public function getPoziom(): ?int
    {
        return $this->poziom;
    }

    public function setPoziom(int $poziom): void
    {
        $this->poziom = $poziom;
    }

    public function getNauczyciel(): ?Nauczyciel
    {
        return $this->nauczyciel;
    }

    public function setNauczyciel(Nauczyciel $nauczyciel): void
    {
        $this->nauczyciel = $nauczyciel;
    }

    /**
     * @return Collection|Uczen[]
     */
    public function getUczniowie(): Collection
    {
        return $this->uczniowie;
    }

    public function addUczen(Uczen $uczen): void
    {
        if (!$this->uczniowie->contains($uczen)) {
            $this->uczniowie[] = $uczen;
            $uczen->setKlasa($this);
        }
    }

    public function removeUczen(Uczen $uczen): void
    {
        if ($this->uczniowie->removeElement($uczen)) {
            if ($uczen->getKlasa() === $this) {
                $uczen->setKlasa(null);
            }
        }
    }

    /**
     * @return Collection|Przedmiot[]
     */
    public function getPrzedmioty(): Collection
    {
        return $this->przedmioty;
    }

    public function addPrzedmiot(Przedmiot $przedmiot): void
    {
        if (!$this->przedmioty->contains($przedmiot)) {
            $this->przedmioty[] = $przedmiot;
            $przedmiot->setKlasa($this);
        }
    }

    public function removePrzedmiot(Przedmiot $przedmiot): void
    {
        if ($this->przedmioty->removeElement($przedmiot)) {
            if ($przedmiot->getKlasa() === $this) {
                $przedmiot->setKlasa(null);
            }
        }
    }
}
?>